fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AbstractRolesAuthorizationDelegate.java

```diff
@@ -62,18 +63,13 @@
      * @param principals effective principals
      * @return set of effective content roles
      */
-    public static Set<String>
-    resolveUserRoles(final Map<String, List<String>> acl,
-                    final Set<Principal> principals) {
-        final Set<String> roles = new HashSet<>();
-        for (final Principal p : principals) {
-            final List<String> matchedRoles = acl.get(p.getName());
-            if (matchedRoles != null) {
-                LOGGER.debug("request principal matched role assignment: {}", p.getName());
-                roles.addAll(matchedRoles);
-            }
-        }
-        return roles;
+    public static Set<String> resolveUserRoles(final Map<String, Collection<String>> acl,
+                    final Collection<Principal> principals) {
+        return principals.stream().map(Principal::getName).filter(acl::containsKey)
```

Comment:
This is just a question (trying to better my stream knowledge).

Is this filtering out any acl that already exists in the stream?